api/app/tests/apps/api/v1/test_public_question_endpoint.py

```diff
@@ -0,0 +1,45 @@
+import os
+
+from tests.apps.signals.factories import CategoryFactory, ParentCategoryFactory, QuestionFactory
+from tests.test import SignalsBaseApiTestCase
+
+THIS_DIR = os.path.dirname(__file__)
+
+
+class TestCategoryQuestionEndpoints(SignalsBaseApiTestCase):
+    def setUp(self):
+        self.retrieve_sub_category_question_schema = self.load_json_schema(
+            os.path.join(
+                THIS_DIR,
+                'json_schema',
+                'get_signals_v1_public_questions_categories_{slug}_sub_categories_{sub_slug}.json'
+            )
+        )
+
+        self.question = QuestionFactory.create_batch(1)
+        self.parent_category = ParentCategoryFactory.create()
+        CategoryFactory.create_batch(1, parent=self.parent_category, questions=self.question)
+        self.parent_category.refresh_from_db()
+        super(TestCategoryQuestionEndpoints, self).setUp()
+
+    def test_category_question_list(self):
+        endpoint_url = '/signals/v1/public/questions/'
+        response = self.client.get(endpoint_url)
+        self.assertEqual(response.status_code, 200)
+        data = response.json()
+        # JSONSchema validation
+        self.assertJsonSchema(self.retrieve_sub_category_question_schema, data)
+        self.assertEqual(data['count'], 1)
+
+        # filter on main and sub
```

Comment:
It would be nice to also have a test for filtering only on only `main_slug` main category, with two questions in the database, one that has that main category and no sub category and one with that main category and a sub category. (The latter should not be returned as I read it.)

No blocker though.

Reply:
I see you added it, great!